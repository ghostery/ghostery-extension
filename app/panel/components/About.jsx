/**
 * About Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */
import React from 'react';
import globals from '../../../src/classes/Globals';
import { sendMessage } from '../utils/msg';

const { BROWSER_INFO, EXTENSION_VERSION } = globals;
/**
 * @class Implement About view which opens from the main drop-down menu.
 * @memberof PanelClasses
 */
class About extends React.Component {
	/**
	 * Open internal Licenses page with licenses of
	 * all third-party packages used by Ghostery.
	 * @static
	 */
	static openNewTab() {
		sendMessage('openNewTab', {
			url: chrome.runtime.getURL('./app/templates/licenses.html'),
			become_active: true,
		});
		window.close();
	}
	/**
	 * Render About panel.
	 * @return {ReactComponent}   ReactComponent instance
	 */
	render() {
		return (
			<div id="content-about">
				<div className="row">
					<div className="small-12 columns">
						<h1>{ t('panel_about_panel_header') }</h1>
						<div className="support-section">
							<h3>{ t('panel_about_version_header', [BROWSER_INFO.displayName, EXTENSION_VERSION]) }</h3>
							<a href="https://github.com/ghostery/ghostery-extension/releases" target="_blank" rel="noopener noreferrer">
								{ t('panel_about_release_notes') }
							</a>
							<a href="https://www.mozilla.org/en-US/MPL/2.0/" target="_blank" rel="noopener noreferrer">
								{ t('panel_about_license') }
							</a>
							<a href="https://www.ghostery.com/about-ghostery/browser-extension-privacy-policy/" target="_blank" rel="noopener noreferrer">
								{ t('panel_about_privacy_statement') }
							</a>
							<div onClick={About.openNewTab}>
								{ t('panel_about_licenses') }
							</div>
							<a href="https://www.ghostery.com/" target="_blank" rel="noopener noreferrer">Ghostery.com</a>
						</div>
					</div>
				</div>
			</div>
		);
	}
}

export default About;
