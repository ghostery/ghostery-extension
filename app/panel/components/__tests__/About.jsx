/**
 * About Test Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2019 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React from 'react';
import renderer from 'react-test-renderer';
import About from '../About';

describe('app/panel/components/About.jsx', () => {
	describe('Snapshot tests with react-test-renderer', () => {
		test('About panel is rendered correctly', () => {
			const component = renderer.create(
				<About />
			).toJSON();
			expect(component).toMatchSnapshot();
		});
	});
});
