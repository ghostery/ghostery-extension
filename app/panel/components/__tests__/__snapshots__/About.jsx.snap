// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/panel/components/About.jsx Snapshot tests with react-test-renderer About panel is rendered correctly 1`] = `
<div
  id="content-about"
>
  <div
    className="row"
  >
    <div
      className="small-12 columns"
    >
      <h1>
        panel_about_panel_header
      </h1>
      <div
        className="support-section"
      >
        <h3>
          panel_about_version_header
        </h3>
        <a
          className=""
          href="https://github.com/ghostery/ghostery-extension/releases"
          onClick={[Function]}
        >
          panel_about_release_notes
        </a>
        <a
          className=""
          href="https://www.mozilla.org/en-US/MPL/2.0/"
          onClick={[Function]}
        >
          panel_about_license
        </a>
        <a
          className=""
          href="https://www.ghostery.com/about-ghostery/browser-extension-privacy-policy/"
          onClick={[Function]}
        >
          panel_about_privacy_statement
        </a>
        <a
          className=""
          href="https://www.ghostery.com/about-ghostery/ghostery-terms-and-conditions/"
          onClick={[Function]}
        >
          panel_about_terms_and_conditions
        </a>
        <a
          className=""
          href="https://www.ghostery.com/about-ghostery/imprint/"
          onClick={[Function]}
        >
          panel_about_imprint
        </a>
        <a
          className=""
          onClick={[Function]}
        >
          panel_about_licenses
        </a>
        <a
          className=""
          href="https://www.ghostery.com/"
          onClick={[Function]}
        >
          Ghostery.com
        </a>
      </div>
    </div>
  </div>
</div>
`;
